Source/ASNetworkImageNode.mm

```diff
@@ -277,6 +281,24 @@ - (void)_setCurrentImageQuality:(CGFloat)imageQuality
   });
 }
 
+- (void)setLoadingProgress:(CGFloat)loadingProgress
+{
+  ASLockScopeSelf();
+  _loadingProgress = loadingProgress;
+}
+
+- (CGFloat)loadingProgress
+{
+  return ASLockedSelf(_loadingProgress);
+}
+
+- (void)_locked_setLoadingProgress:(CGFloat)loadingProgress
```

Comment:
Let's add a explanation why we're dispatching to the main queue here instead of locking. Something similar to what `_setCurrentImageQuality` has.